src/Microsoft.AspNet.Cors.Core/CorsEngine.cs

```diff
@@ -0,0 +1,187 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNet.Http;
+using Microsoft.Framework.Internal;
+using Microsoft.Framework.OptionsModel;
+
+namespace Microsoft.AspNet.Cors.Core
+{
+    /// <summary>
+    /// Default implementation of <see cref="ICorsEngine"/>.
+    /// </summary>
+    public class CorsEngine : ICorsEngine
```

Comment:
rename `CorsService : ICorsService` to match pattern like `AuthorizationService : IAuthorizationService`